collections/_article/dev-snapshot-godot-4-4-dev-4.md

```diff
@@ -0,0 +1,191 @@
+</video>
+
+
+### Ubershaders and pipeline pre-compilation (and dedicated transfer queues)
+
+This change combines 2 huge improvements ([GH-90400](https://github.com/godotengine/godot/pull/90400)):
+1. Ubershaders: a technique that allows us to completely avoid shader stutter when objects are viewed in a scene for the first time
+2. Dedicated transfer queues: by relying on special hardware functions we can dramatically improve the load time for scenes containing many graphics assets like meshes and textures.
+
+Thanks to [DarioSamo](https://github.com/dariosamo) for months of hard work on this. The implementation of Ubershaders
+is something that took a lot of design and implementation to get right, but the end result is something we are very
+happy with.
+
+“Ubershaders” are a version of each shader that contains all features and are compiled at load time. We pair the
+ubershader with the ability to compile specialized shader pipelines in the background. This way when we draw a new
+object, instead of freezing and waiting for the pipeline to compile, we fallback on the ubershader and compile the
+specialized shader in the background.
+
+Most games will notice the improvement from ubershaders without any content changes. However, there are certain cases
+where the ubershader won't work and some content changes will be needed to avoid shader stutter. However, the
+workarounds for avoiding shader stutter are now very simple and are well documented already in the [official docs](https://docs.godotengine.org/en/latest/tutorials/performance/pipeline_compilations.html).
+
+Taking advantage of the new ubershader infrastructure, we can now optimize our shaders by creating specialized variants
+that skip unused pathways. We have already begin doing so to optimize our existing shaders
+([GH-98825](https://github.com/godotengine/godot/pull/98825)).
+
+### Add Shadow Caster Mask property to Light3D
+
+This long awaited improvement from [EMBYRDEV](https://github.com/EMBYRDEV) allows users to apply a mask on Light3Ds to
+select what rendering layers will be considered when casting shadows
+([GH-85338](https://github.com/godotengine/godot/pull/85338)). Previously it was only possible to disable shadows from a
+GeometryInstance (for all Light3Ds) or a Light3D (for all GeometryInstances). This allows much more fine-grained control
+that allows users to further optimize dynamic lights and control where shadows appear in their games. 
+
+### Always add decimal when converting float to string 
+
+Currently when printing float values Godot trims trailing `.0`s as the are technically unnecessary. However, this could
+lead to issues for users who need to distinguish between floating point numbers and integers.
+[KoBeWi](https://github.com/kobewi) has now changed the behavior when floats are converted to strings so that the
+trailing `.0` is always printed when floats are converted to strings ([GH-47502](https://github.com/godotengine/godot/pull/47502)). 
```

Comment:
```suggestion
[KoBeWi](https://github.com/kobewi) has now changed the behavior when floats are converted to strings to always add the decimal number, even if it would be `.0`.
This makes floats always recognisable when they are converted to strings ([GH-47502](https://github.com/godotengine/godot/pull/47502)). 
```
Or similar

Reply:
```suggestion
[KoBeWi](https://github.com/kobewi) has now changed the behavior when floats are converted to strings to always add the decimal number, even if it would be `.0`.
This makes floats always recognizable when they are converted to strings ([GH-47502](https://github.com/godotengine/godot/pull/47502)). 
```

Looks good, small change to use US spelling instead of UK